### ISSUE TITLE
turbo/examples/with-tailwind tailwind prefix ("ui-") not working without building ui lib

### ISSUE THREAD

**Issue by mule-stand** [vercel/turbo#5755]
### What version of Turborepo are you using?

latest

### What package manager are you using / does the bug impact?

pnpm

### What operating system are you using?

Linux

### Describe the Bug

It`s not working https://github.com/vercel/turbo/tree/main/examples/with-tailwind#building-packagesui
Without assembly, the prefix for the ui library does not rebate and as a consequence styles are not applied when using the prefix in className

### Expected Behavior

Prefixes must be worked

### To Reproduce

1. Uncomment line https://github.com/vercel/turbo/blob/b1b74d78de1d1a6d8dce9f1f98e768c8274023f2/examples/with-tailwind/packages/tailwind-config/tailwind.config.js#L8
2. This https://github.com/vercel/turbo/blob/b1b74d78de1d1a6d8dce9f1f98e768c8274023f2/examples/with-tailwind/packages/ui/package.json#L8C3-L11C5 replace by this:
```json
 "exports": {
    ".": "./src/index.tsx",
    "./styles.css": "./src/styles.css"
  }
```

**Comment by tknickman** [comment]
This is fixed as part of https://github.com/vercel/turbo/pull/5912

**Comment by JKarlavige** [comment]
Hey @tknickman, i'm running into the same issue. When not bundling the UI package, the Tailwind prefixes do not work. In VSCode, it picks up the `ui-` prefix which I have set in the `ui` package's `tailwind.config.js`:

```
import sharedConfig from 'web-tailwind'

const config = {
  presets: [sharedConfig],
  prefix: 'ui-',
}
```
![image](https://github.com/vercel/turbo/assets/13647110/dfd04dcf-7508-45b1-af1a-f767e69115d3)

However, when importing a component into an app, the prefix classes are not applied. I've spent the day looking into it and haven't figured out a solution. Is this not possible without bundling the ui package?
